### PR TITLE
feat(ui5-input): implement showClearIcon property

### PR DESCRIPTION
**Pull request by MapTo0**
FIXES: https://github.com/SAP/ui5-webcomponents/issues/4395

### CODE REVIEW

File: packages/main/src/Input.js [hunk 1]
```diff
@@ -330,6 +330,30 @@ const metadata = {
 			defaultValue: "",
 		},
 
+		/**
+		 * Defines whether the clear icon of the input will be shown.
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 * @since 1.1.2
```
Comment (ilhan007) [inline packages/main/src/Input.js:339]: 1.1.2 is already produced, you can set 1.2.0

File: packages/main/test/specs/Input.spec.js [hunk 2]
```diff
@@ -636,6 +636,77 @@ describe("Input general interaction", () => {
...
+		// press clear icon
+		await clearIcon.click();
+
+		assert.strictEqual(await input.getProperty("value"), "", "Clear icon clear the value");
+		assert.notOk(await input.getProperty("effectiveShowClearIcon"), "Clear icon should not be shown");
+		assert.strictEqual(await changeCounter.getText(), "0", "Change event not called yet");
+		assert.strictEqual(await inputCounter.getText(), "2", "Input event called when typing or clear action is done");
+	});
+
+	it("Change event is called when value of inut is cleared with clear icon and input is focused out", async () => {
```
Comment (ilhan007) [inline packages/main/test/specs/Input.spec.js:669]: inut-> input

File: packages/main/test/specs/Input.spec.js [hunk 3]
```diff
@@ -636,6 +636,77 @@ describe("Input general interaction", () => {
...
+		await innerInput.click();
+		await innerInput.keys("a");
+		await changeCounter.click();
+
+		const clearIcon = await input.shadow$(".ui5-input-clear-icon");
+
+		// press clear icon
+		await clearIcon.click();
+
+		assert.strictEqual(await changeCounter.getText(), "2", "Change event not called twice");
```
Comment (ilhan007) [inline packages/main/test/specs/Input.spec.js:687]: It says "not called twice", but compared with 2, a bit confusing when reading trough it.

**Review by ilhan007 — CHANGES_REQUESTED** [review]
Did not test it, but the code looks fine, some minor changes needed. Perhaps the team would test it anyway.

File: packages/main/test/specs/Input.spec.js [hunk 4]
```diff
@@ -684,7 +684,7 @@ describe("Input general interaction", () => {
 		// press clear icon
 		await clearIcon.click();
 
-		assert.strictEqual(await changeCounter.getText(), "2", "Change event not called twice");
+		assert.strictEqual(await changeCounter.getText(), "2", "Change event called twice (first - typing, secnond - clear icon)");
```
Comment (ilhan007) [inline packages/main/test/specs/Input.spec.js:687]: Marti, don't want to be mean :D - but "secnond" -> "second"

**Review by niyap — COMMENTED** [review]
From my side, it looks good :) Just one question, do we expected when we have ui5-input with suggestions and a dialog is open on mobile device, to have the clear icon by default, because currently that happens and I am not sure whether it is done by reason or it is a bug.

File: packages/main/test/pages/Input.html [hunk 5]
```diff
@@ -363,6 +363,27 @@ <h3>sap_horizon</h3>
 	<br>
 	<br>
 
+	<div>
+		<h3>Inputs with clear icon</h3>
+
+		<ui5-input id="clear-input" show-clear-icon placeholder="Search for a country ..."></ui5-input>
```
Comment (stbodurov) [inline packages/main/test/pages/Input.html:369]:
```suggestion
		<ui5-input id="clear-input" show-clear-icon placeholder="Type to show clear icon..."></ui5-input>
```

File: packages/main/test/pages/Input.html [hunk 6]
```diff
@@ -363,6 +363,27 @@ <h3>sap_horizon</h3>
...
+		<br />
+		<br />
+		<span>Change: </span><span id="clear-input-change-event-count">0</span>
+		<br />
+		<span>Input: </span><span id="clear-input-input-event-count">0</span>
+		<br />
+		<br />
+		<span class="sapUiSizeCompact">
+			<ui5-input id="clear-input-compact" show-clear-icon placeholder="Search for a country ..." value="test"></ui5-input>
```
Comment (stbodurov) [inline packages/main/test/pages/Input.html:378]:
```suggestion
			<ui5-input id="clear-input-compact" show-clear-icon placeholder="Type to show clear icon..." value="test"></ui5-input>
```

**Review by stbodurov — COMMENTED** [review]
When you open an input with suggestions on mobile and set a value, the clear icon does not appear in the main input once you close the dialog. This is not the case in UI5.

File: packages/main/src/InputPopover.hbs [hunk 7]
```diff
@@ -24,15 +24,15 @@
 			</div>
 			<div class="row">
 				<div class="input-root-phone">
-					<input
+					<ui5-input
 						class="ui5-input-inner-phone"
 						type="{{inputType}}"
 						.value="{{value}}"
-						inner-input
+						.show-clear-icon={{showClearIcon}}
```
Comment (stbodurov) [inline packages/main/src/InputPopover.hbs:31]: Now what happens is the opposite - the clear icon is forwarded to the main input, but no longer shows in the dialog.

Removing the period seems to bind the property correctly:

```suggestion
						show-clear-icon={{showClearIcon}}
```

**Review by stbodurov — COMMENTED** [review]
We can also enable `show-clear-icon` for the "**Input With Suggestions**" sample in the playground:

packages/main/test/samples/Input.sample.html:L49

**Review by stbodurov — APPROVED** [review]

**Pull request merged**